cmd/k8s-operator/deploy/chart/templates/deployment.yaml

```
@@ -34,17 +34,11 @@ spec:
       securityContext:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if or .Values.oauth.clientSecret .Values.oauth.audience }}
       volumes:
-        {{- if .Values.oauth.clientSecret }}
+        {{- if .Values.oauthSecretVolume }}
```

Hard to apply the suggestion cleanly, but I think this works with less churn:

```suggestion
      {{- if or .Values.oauth.clientSecret .Values.oauth.audience .Values.oauthSecretVolume }}
      volumes:
        {{- if or .Values.oauth.clientSecret .Values.oauthSecretVolume }}
```

Tested with:

```
helm template . --show-only templates/deployment.yaml
helm template . --show-only templates/deployment.yaml --set oauthSecretVolume.csi=true
helm template . --show-only templates/deployment.yaml --set oauth.clientSecret=foo
helm template . --show-only templates/deployment.yaml --set oauth.audience=foo
```